### PR TITLE
add LMI feature matrix

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Feature matrix, better to check if there are some missing columns.
@siddvenk

### CODE REVIEW

File: serving/docs/lmi_new/deployment_guide/README.md [hunk 1]
```diff
@@ -57,3 +57,19 @@ These libraries expose features and capabilities through different APIs and mech
...
+| support quantization                  | BitsandBytes/GPTQ      | GPTQ/AWQ         | SmoothQuant INT8   | SmoothQuant, AWQ, GPTQ | INT8                         | GPTQ/AWQ     |
+| GPU machine supported                 | G4/G5/P4D/P5           | G5/P4D/P5        | G5/P4D/P5          | G5/P4D/P5              | INF2/TRN1                    | G4/G5/P4D/P5 |
+| execution mode                        | Python                 | MPI              | MPI                | MPI                    | Python                       | Python       |
+| multi-accelerator weight loading      | Yes                    | Yes              | Yes                | Yes                    | Yes                          | Yes          |
+| tensor parallel                       | No                     | Yes              | Yes                | Yes                    | Yes                          | Yes          |
+| continuous batching streaming         | Yes                    | Yes              | Yes                | Yes                    | Yes                          | Yes          |
```
Comment (siddvenk) [inline serving/docs/lmi_new/deployment_guide/README.md:72]: should we split continuous batching and streaming into two separate features? streaming can be toggled independently of continuous batching.

**Comment by siddvenk** [comment]
Some other features we may consider:
- PagedAttention/FlashAttention kernels?
- MQA/GQA

**Comment by lanking520** [comment]
> Some other features we may consider:
> 
> * PagedAttention/FlashAttention kernels?
> * MQA/GQA

This is customer facing doc, not sure if we want to show this as comparison

File: serving/docs/lmi_new/deployment_guide/README.md (hunk 1 above)
Comment (siddvenk) [reply serving/docs/lmi_new/deployment_guide/README.md:72]: It's possible to disable streaming when using continuous batching, right?

Comment (lanking520) [reply serving/docs/lmi_new/deployment_guide/README.md:72]: customer can always choose to send back the content when it received full, this is the contract with SageMaker. So SageMaker customer will feel like both supported.

But at LMI layer, we always send back token-by-token

Comment (lanking520) [reply serving/docs/lmi_new/deployment_guide/README.md:72]: so depending on how client plan to present it

File: serving/docs/lmi_new/deployment_guide/README.md [hunk 2]
```diff
@@ -57,3 +57,19 @@ These libraries expose features and capabilities through different APIs and mech
...
+
+## Feature Matrix
+
+|                                       | HuggingFace Accelerate | LMI_dist (8.0.0) | DeepSpeed (0.12.6) | TensorRTLLM (0.7.1)    | TransformersNeuronX (2.16.0) | vLLM (0.2.7) |
+| ------------------------------------- | ---------------------- | ---------------- | ------------------ | ---------------------- | ---------------------------- | ------------ |
+| DLC                                   | DeepSpeed              | DeepSpeed        | DeepSpeed          | LMI TRTLLM             | LMI Neuron                   | DeepSpeed    |
+| current handler                       | huggingface            | huggingface      | deepspeed          | tensorrtllm            | transformersneuronx          | huggingface  |
+| support quantization                  | BitsandBytes/GPTQ      | GPTQ/AWQ         | SmoothQuant INT8   | SmoothQuant, AWQ, GPTQ | INT8                         | GPTQ/AWQ     |
+| GPU machine supported                 | G4/G5/P4D/P5           | G5/P4D/P5        | G5/P4D/P5          | G5/P4D/P5              | INF2/TRN1                    | G4/G5/P4D/P5 |
```
Comment (rohithkrn) [inline serving/docs/lmi_new/deployment_guide/README.md:68]: nit: Should this just be `Instance type`? (inf2 is not a gpu?)

File: serving/docs/lmi_new/deployment_guide/README.md [hunk 3]
```diff
@@ -57,3 +57,19 @@ These libraries expose features and capabilities through different APIs and mech
...
+| execution mode                        | Python                 | MPI              | MPI                | MPI                    | Python                       | Python       |
```
Comment (sean-eich) [inline serving/docs/lmi_new/deployment_guide/README.md:69]: For MPI and Python please link to the explanations of the two execution modes.

File: serving/docs/lmi_new/deployment_guide/README.md (hunk 1 above)
Comment (sean-eich) [reply serving/docs/lmi_new/deployment_guide/README.md:72]: Can you please elaborate on what the SageMaker customer would need to do in order to "toggle" streaming?  How, specifically, are they waiting to receive the full content?  Is this a configuration on SM side or a manual implementation for the user?  Whatever the case, lets call out as a footnote what SageMaker customers can do to toggle.

Comment (lanking520) [reply serving/docs/lmi_new/deployment_guide/README.md:72]: @sean-eich It is not a container functionalties for customer to do streaming or not. The container is forever sending streaming content and SageMaker will receive them into:
- Standard output: Non streaming, use could just do `predictor.predict` to get a full json string
- Received as token outputs: In streaming fashion

There is nothing we can do at LMI level to make it happen, it is all depends on how customer use SageMaker client

File: serving/docs/lmi_new/deployment_guide/README.md (hunk 3 above)
Comment (lanking520) [reply serving/docs/lmi_new/deployment_guide/README.md:69]: @siddvenk has a Conceptural guide for it. We can add link to this once that doc is ready

File: serving/docs/lmi_new/deployment_guide/README.md (hunk 2 above)
Comment (lanking520) [reply serving/docs/lmi_new/deployment_guide/README.md]: addressed

File: serving/docs/lmi_new/deployment_guide/README.md [hunk 4]
```diff
@@ -57,3 +57,19 @@ These libraries expose features and capabilities through different APIs and mech
 With LMI's built-in handlers, there is no need to learn each library and write custom code to leverage the features and optimizations they offer.
 We expose a unified configuration format that allows you to easily switch between libraries as they evolve and improve over time.
 As the ecosystem grows and new libraries become available, LMI can integrate them and offer the same consistent experience.
+
+## Feature Matrix
+
+|                                       | HuggingFace Accelerate | LMI_dist (8.0.0) | DeepSpeed (0.12.6) | TensorRTLLM (0.7.1)    | TransformersNeuronX (2.16.0) | vLLM (0.2.7) |
+| ------------------------------------- | ---------------------- | ---------------- | ------------------ | ---------------------- | ---------------------------- | ------------ |
+| DLC                                   | DeepSpeed              | DeepSpeed        | DeepSpeed          | LMI TRTLLM             | LMI Neuron                   | DeepSpeed    |
+| current handler                       | huggingface            | huggingface      | deepspeed          | tensorrtllm            | transformersneuronx          | huggingface  |
```
[the remark this answers is not in the document]
Comment (lanking520) [reply serving/docs/lmi_new/deployment_guide/README.md]: addressed with links

File: serving/docs/lmi_new/deployment_guide/README.md (hunk 1 above)
Comment (sean-eich) [reply serving/docs/lmi_new/deployment_guide/README.md]: Understood.  Perhaps we can just add a footnote with a link to docs that detail how SM customers can work with the container's streaming.  The idea being that we want to make it super simple for customers to find all the information that they need.

**Review by rohithkrn — APPROVED** [review]

**Pull request merged**